src/main/java/io/jenkins/plugins/coverage/model/CoverageStatistics.java

```diff
@@ -0,0 +1,151 @@
+        return getPercentage(QualityGate.QualityGateType.METHOD.getName());
+    }
+
+    /**
+     * Returns the percentage the line coverage.
+     *
+     * @return the percentage of the line coverage
+     */
+    public double getLinePercentage() {
+        return getPercentage(QualityGate.QualityGateType.LINE.getName());
+    }
+
+    /**
+     * Returns the percentage the package coverage.
+     *
+     * @return the percentage of the package coverage
+     */
+    public double getPackagePercentage() {
+        return getPercentage(QualityGate.QualityGateType.PACKAGE.getName());
+    }
+
+    /**
+     * Returns the percentage the branch coverage.
+     *
+     * @return the percentage of the branch coverage
+     */
+    public double getBranchPercentage() {
+        return getPercentage(QualityGate.QualityGateType.BRANCH.getName());
+    }
+
+    /**
+     * Returns the percentage the class coverage.
+     *
+     * @return the percentage of the class coverage
+     */
+    public double getClassPercentage() {
+        return getPercentage(QualityGate.QualityGateType.CLASS.getName());
+
+    }
+
+    /**
+     * Returns the percentage the instruction coverage.
+     *
+     * @return the percentage of the instruction coverage
+     */
+    public double getInstructionPercentage() {
+        return getPercentage(QualityGate.QualityGateType.INSTRUCTION.getName());
+    }
+
+    /**
+     * Returns the covered percentage by key.
+     *
+     * @param key the key
+     * @return the covered percentage
+     */
+    private double getPercentage(final String key) {
+        return sortedCoverages.containsKey(key)
+                ? sortedCoverages.get(key).getCoveredPercentage() : 0;
+    }
+
+    /**
+     * Reference for the method to get the covered percentage.
+     */
+    public enum StatisticProperties {
+        FILE(CoverageStatistics::getFilePercentage),
+        MODULE(CoverageStatistics::getModulePercentage),
+        METHOD(CoverageStatistics::getMethodPercentage),
+        LINE(CoverageStatistics::getLinePercentage),
+        PACKAGE(CoverageStatistics::getPackagePercentage),
+        BRANCH(CoverageStatistics::getBranchPercentage),
+        CLASS(CoverageStatistics::getClassPercentage),
+        INSTRUCTION(CoverageStatistics::getInstructionPercentage);
+
+        private final SerializableGetter coverageGetter;
+
+        /**
+         * Ctor for StatisticProperties.
+         *
+         * @param coverageGetter The coverage getter
+         */
+        StatisticProperties(final SerializableGetter coverageGetter) {
+            this.coverageGetter = coverageGetter;
+        }
+
+        public Function<CoverageStatistics, Double> getCoverageGetter() {
+            return coverageGetter;
+        }
+    }
+
+    /**
+     * Interface for SerializableGetter.
+     */
+    private interface SerializableGetter extends Function<CoverageStatistics, Double>, Serializable {
+        // nothing to add
+    }
+}
```

Comment:
This feels a little bit like an overkill for the actual specification in the issue. It also does not allow different coverage metrics that are not defined. 

But in general this is exactly the way I need this code in practice to use it in the user interface as well. 👍